api/filters/replacement/replacement.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package replacement
+
+import (
+	"fmt"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/provider"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+type Filter struct {
+	Replacements []types.Replacement
+}
+
+var _ kio.Filter = Filter{}
+
+// Filter replaces values of targets with values from sources
+// TODO (#3492): Connect this to a replacement transformer plugin
+func (f Filter) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	var result []*yaml.RNode
+	rf := provider.NewDefaultDepProvider().GetResourceFactory()
+	m, err := resmap.NewFactory(rf).NewResMapFromRNodeSlice(nodes)
+	if err != nil {
+		return nil, err
+	}
+	for i := range nodes {
+		r, err := applyReplacements(f.Replacements, nodes[i], m)
+		if err != nil {
+			return nil, err
+		}
+		if r != nil {
```

Comment:
This made me wonder if we're dropping nodes that don't get substituted, but looking into `applyReplacements`, it seems like this check isn't necessary.... i.e. when would `r` be `nil` and `err` above not be?